test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -216,9 +216,27 @@ public enum ProductFeature {
     }
 
     private static EnumSet<ProductFeature> availableFeatures;
-    private static Set<String> nodeVersions;
+    private static Set<String> nodesVersions;
     private static TestFeatureService testFeatureService;
 
+    protected static Set<String> getCachedNodeVersions() {
```

Comment:
This should be plural (Node**s** ) as well. 
Nit, how about just using `nodesVersions` to not make this look like a regular getter?